lib/logger/lib/logger.ex

```diff
@@ -19,6 +19,32 @@ defmodule Logger do
     * Wraps OTP's `error_logger` to prevent it from
       overflowing.
 
+  Logging is useful for tracking when an event of interest happens in your
+  system. For example, logging incoming synchronous calls in a `GenServer`
+  behaviour can provide valuable insight.
+
+      def handle_call(request, from, state) do
+        Logger.info fn ->
+          "handle_call/3: #{inspect request} request received " <>
+          "from #{inspect from} with state #{inspect state}."
+        end
+        # ...
+      end
```

Comment:
I would keep this example more abstract (since there are tools to inspect messages and calls in GenServer and other OTP behaviours). What about something very abstract like:

```elixir
def delete_user(user) do
  Logger.info fn ->
    "Deleting user from the system: #{inspect(user)}"
  end
end
```

?